PortalSwift/Sources/PortalSwift/Storage/GDrive/GoogleAuth.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  GoogleAuth.swift
+//  PortalSwift
+//
+//  Created by Blake Williams on 2/7/23.
+//
+
+import Foundation
+import GoogleSignIn
+import UIKit
+
+public class GoogleAuth {
+  public var auth: GIDSignIn
+  public var config: GIDConfiguration
+  public var view: UIViewController?
+
+  init(config: GIDConfiguration, view: UIViewController? = nil) {
+    self.auth = GIDSignIn.sharedInstance
+    self.config = config
+    self.view = view
+  }
+
+  func getAccessToken() async -> String {
+    do {
+      if self.hasPreviousSignIn() {
+        // Attempt to sign in silently
+        let user = try await self.restorePreviousSignIn()
+        return user.authentication.accessToken
+      } else {
+        // User has not signed in before, prompt for sign-in
+        let user = try await self.signIn()
+        return user.authentication.accessToken
+      }
+    } catch {
+      return ""
+    }
+  }
+
+  func getCurrentUser() -> GIDGoogleUser? {
+    self.auth.currentUser
+  }
+
+  func hasPreviousSignIn() -> Bool {
+    self.auth.hasPreviousSignIn()
+  }
+
+  func restorePreviousSignIn() async throws -> GIDGoogleUser {
+    let user = try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<GIDGoogleUser, Error>) in
+      self.auth.restorePreviousSignIn { user, error in
+        if error != nil {
```

Comment:
should this be 
```suggestion
        if let error = error  {
            return continuation.resume(throwing: error) 
```
to avoid using the ! operator

Reply:
Either works, but they accomplish the same thing.